### PR TITLE
703-Add db_connection param to POST Workspace Layers

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #703

- [x] Tests
- ~~Layman Test Client~~
- [x] Changelog
- [x] Documentation

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: test_tools/process_client.py [hunk 1]
```diff
@@ -326,11 +327,12 @@ def publish_workspace_publication(publication_type,
 
     # map layers must not be set together with file_paths
```
Comment (jirik) [inline test_tools/process_client.py:328]:
```suggestion
```

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/layer/rest_workspace_layers.py [hunk 2]
```diff
@@ -54,6 +54,21 @@ def post(workspace):
         ]
     input_files = fs_util.InputFiles(sent_streams=sent_file_streams, sent_paths=sent_file_paths)
     db_connection = request.form.get('db_connection', '')
+    if len(input_files.raw_paths) == 0 and not db_connection:
+        raise LaymanError(1, {
```
[the remark this answers is not in the document]
Comment (index-git) [reply src/layman/layer/rest_workspace_layers.py:58]: I think, that 1-"Missing parameter" is better.

File: doc/rest.md [hunk 3]
```diff
@@ -88,9 +88,9 @@ Have the same request parameters and response structure and headers as [GET Laye
 Publish vector or raster data file as new layer of WMS, in case of vector also new feature type of WFS.
 
 Processing chain consists of few steps:
-- save file to workspace directory within Layman data directory
+- save eventual files to workspace directory within Layman data directory
```
Comment (jirik) [inline doc/rest.md:91]:
```suggestion
- save files (if sent) to workspace directory within Layman data directory
```

File: doc/rest.md [hunk 4]
```diff
@@ -88,9 +88,9 @@ Have the same request parameters and response structure and headers as [GET Laye
...
 - save basic information (name, title, access_rights) into PostgreSQL
-- for vector layers import the vector file to PostgreSQL database as new table into workspace schema
+- for vector layers import eventual vector file to PostgreSQL database as new table into workspace schema
```
Comment (jirik) [inline doc/rest.md:93]:
```suggestion
- for vector layers import vector file (if sent) to PostgreSQL database as new table into workspace schema
```

File: doc/rest.md [hunk 5]
```diff
@@ -121,7 +121,7 @@ Check [Asynchronous file upload](async-file-upload.md) example.
 Content-Type: `multipart/form-data`, `application/x-www-form-urlencoded`
 
 Body parameters:
-- **file**, file(s) or file name(s)
+- *file*, file(s) or file name(s)
```
Comment (jirik) [inline doc/rest.md:124]:
```suggestion
- *file*, file(s) or file name(s)
  - exactly one of `file` or `db_connection` must be set
```

File: doc/rest.md [hunk 6]
```diff
@@ -143,6 +143,8 @@ Body parameters:
    - if published file has empty bounding box (i.e. no features), its bounding box on WMS/WFS endpoint is set to the whole World
    - attribute names are [laundered](https://gdal.org/drivers/vector/pg.html#layer-creation-options) to be safely stored in DB
    - if QML style is used in this request, it must list all attributes contained in given data file
+- *db_connection*, string
+  - is not allowed in combination with *file* parameter
```
Comment (jirik) [inline doc/rest.md:147]:
```suggestion
  - exactly one of `file` or `db_connection` must be set
```

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/db/util.py [hunk 7]
```diff
@@ -104,3 +104,16 @@ def ensure_srid_definition(srid, proj4text):
 ON CONFLICT (srid) DO UPDATE SET proj4text = %s;'''
     params = (srid, proj4text, proj4text)
     run_statement(sql, params)
+
+
+def parse_connection_string(connection_string):
+    from urllib import parse
```
Comment (jirik) [inline src/db/util.py:110]: Move import up

File: src/db/util.py [hunk 8]
```diff
@@ -104,3 +104,16 @@ def ensure_srid_definition(srid, proj4text):
...
+    connection = parse.urlparse(connection_string, )
+    params = parse.parse_qs(connection.query)
+    url = connection._replace(query='').geturl()
+    result = {
+        'url': url,
```
Comment (jirik) [inline src/db/util.py:115]: Create new dataclass for connection

File: src/db/util_test.py [hunk 9]
```diff
@@ -12,3 +12,25 @@
...
+        'url': 'postgresql://user:password@postgresql:5432/dbname',
+        'table': 'table_name',
+        'geo_column': 'wkb_geometry',
+    }),
+    ('postgresql://postgresql', {
+        'url': 'postgresql://postgresql',
+        'table': None,
+        'geo_column': None,
+    }),
+    ('', {
```
Comment (jirik) [inline src/db/util_test.py:28]: Try also:
- ` `
- `_`
- `$^&*(`
- `ščžýžý`

File: src/db/util.py [hunk 10]
```diff
@@ -104,3 +104,16 @@ def ensure_srid_definition(srid, proj4text):
 ON CONFLICT (srid) DO UPDATE SET proj4text = %s;'''
     params = (srid, proj4text, proj4text)
     run_statement(sql, params)
+
+
+def parse_connection_string(connection_string):
+    from urllib import parse
+    connection = parse.urlparse(connection_string, )
```
Comment (jirik) [inline src/db/util.py:111]: Assert that connection.scheme == "postgresql" and that connection.netloc is not empty. Otherwise raise LaymanError

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — APPROVED** [review]
